tests/test_routes/test_environment_route.py

```diff
@@ -0,0 +1,244 @@
+        with pytest.raises(Exception):
+            response = client.get("/environments/123")
+            assert response.status_code == 500
+            assert response.json() == {"detail": "Failed to read environment: error"}
+
+def test_update_env():
+    with patch.object(
+        EnvController,
+        "update_env_by_id",
+        return_value=EnvUpdated(env_id="123"),
+    ) as mock_update_env:
+        response = client.put(
+            "/environments/123", json={"longitude": 1, "latitude": 1}
+        )
+        assert response.status_code == 200
+        assert response.json() == {
+            "env_id": "123",
+            "message": "Environment successfully updated",
+        }
+        mock_update_env.assert_called_once_with(
+            "123", Env(latitude=1, longitude=1)
+        )
+
+def test_update_env_invalid_input():
+    response = client.put("/environments/123", json={"latitude": "foo", "longitude": "bar"})
+    assert response.status_code == 422
+
+def test_update_env_not_found():
+    with patch.object(
+        EnvController,
+        "update_env_by_id",
+        side_effect=HTTPException(status_code=404, detail="Environment not found"),
+    ) as mock_update_env:
+        response = client.put(
+            "/environments/123", json={"longitude": 1, "latitude": 1}
+        )
+        assert response.status_code == 404
+        assert response.json() == {"detail": "Environment not found"}
+        mock_update_env.assert_called_once_with(
+            "123", Env(latitude=1, longitude=1)
+        )
+
+def test_update_env_server_error():
+    with patch.object(
+        EnvController,
+        "update_env_by_id",
+        side_effect=Exception("error"),
+    ) as mock_update_env:
+        with pytest.raises(Exception):
+            response = client.put(
+                "/environments/123", json={"longitude": 1, "latitude": 1}
+            )
+            assert response.status_code == 500
+            assert response.json() == {"detail": "Failed to update environment: error"}
+
+def test_delete_env():
+    with patch.object(
+        EnvController,
+        "delete_env_by_id",
+        return_value=EnvDeleted(env_id="123"),
+    ) as mock_delete_env:
+        response = client.delete("/environments/123")
+        assert response.status_code == 200
+        assert response.json() == {
+            "env_id": "123",
+            "message": "Environment successfully deleted",
+        }
+        mock_delete_env.assert_called_once_with("123")
+
+def test_delete_env_not_found():
+    with patch.object(
+        EnvController,
+        "delete_env_by_id",
+        return_value=EnvDeleted(env_id="123"),
+    ) as mock_delete_env:
+        response = client.delete("/environments/123")
+        assert response.status_code == 200
+        assert response.json() == {
+            "env_id": "123",
+            "message": "Environment successfully deleted",
+        }
+        mock_delete_env.assert_called_once_with("123")
```

Comment:
Not true. There is no reason to let the client know we didn't had the document he was trying to delete, not only for security reasons but also because it is pointless, if he is trying to delete a file that does not exist in the dB then we can simply respond that the file was deleted since it is logically equivalent